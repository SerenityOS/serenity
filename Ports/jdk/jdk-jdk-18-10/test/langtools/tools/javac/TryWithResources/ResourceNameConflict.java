/*
 * @test  /nodynamiccopyright/
 * @bug 8025113
 * @author sogoel
 * @summary Resource var cannot have same name as local variable
 * @compile/fail/ref=ResourceNameConflict.out -XDrawDiagnostics ResourceNameConflict.java
 */

/**
 * Test methods and their description
 * test1() - negative test - local variable used as test resource
 * test2() - negative test - test resource already defined in an enclosing for statement
 */

public class ResourceNameConflict implements AutoCloseable {

    static final String str = "asdf";

    void test1() {
        String tr = "A resource spec var cannot have same name as local var.";
        try (ResourceNameConflict tr = new ResourceNameConflict()) {
        }
    }

    void test2(String... strArray) {
        for (String str : strArray) {
            try (ResourceNameConflict str = new ResourceNameConflict()) {
            }
        }
    }

    @Override
    public void close() {
    }
}

