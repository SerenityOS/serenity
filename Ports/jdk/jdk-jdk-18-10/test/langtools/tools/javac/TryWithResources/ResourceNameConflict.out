ResourceNameConflict.java:21:35: compiler.err.already.defined: kindname.variable, tr, kindname.method, test1()
ResourceNameConflict.java:27:39: compiler.err.already.defined: kindname.variable, str, kindname.method, test2(java.lang.String...)
2 errors
